lualib/resty/events/worker.lua

```diff
@@ -65,22 +64,28 @@ local function random_delay()
     return random(10, 50) / 1000
 end
 
-local function do_event(self, d)
-    self._callback:do_event(d)
+local function communicate(premature, self)
+    if premature then
+        return
+    end
+
+    self:communicate()
 end
 
 local function start_timer(self, delay)
-    assert(timer_at(delay, function(premature)
-        self:communicate(premature)
-    end))
+    assert(timer_at(delay, communicate, self))
+end
+
+local function terminating(self)
+    return self._connected ~= true or exiting()
```

Comment:
How about `not self._connected`? May be easier to understand.

Reply:
@chronolaw I can change that, but reading `not connected or exiting` always feels like is `not` attached` to `connected` or both `connected and exiting`. But I can change that. Just feeling that ´~= true` makes it more explicit even if you don't have Lua operator order directly in your head.